### PR TITLE
5 create a button component

### PR DESCRIPTION
**Pull request by radicalgraphical**

### CODE REVIEW

File: src/components/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { cn } from "../lib/utils"
+
+type ButtonProps = {
+  variant?: "buttonPrimaryYellowDefault" | "buttonPrimaryYellowDisabled" | "buttonPrimaryYellowSmall" | "buttonSecondaryBlackDefault"| "buttonSecondaryBlackSmallDefault"
+} & React.DetailedHTMLProps<
+  React.ButtonHTMLAttributes<HTMLButtonElement>,
+  HTMLButtonElement
+>
+
```
Comment (ReallyBoard123) [inline src/components/Button.tsx:9]: and then you can even further abstract the tailwind CSS

const variantStyles: Record<ButtonVariant, string> = {
  primary: 'fill your values here',
  secondary: 'fill your values here',
};

same for const sizeStyles and abstract the styling

optional? but you could also put opacity-50 as disabledStyle

File: src/components/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+export function Button({
+  variant = "buttonPrimaryYellowDefault",
+  className,
+  children = "Button",
+  ...props
+}: ButtonProps) {
+  const styles: Record<typeof variant, string> = {
+    buttonPrimaryYellowDefault: "bg-yellow  text-dark-light rounded font font-700 py-4  disabled:opacity-50",
+    buttonPrimaryYellowDisabled: "bg-yellow  text-dark-light rounded font font-700 py-4  opacity-50",
+    buttonPrimaryYellowSmall: "bg-yellow text-dark-light rounded font font-500 py-1  disabled:opacity-50",
```
Comment (ReallyBoard123) [inline src/components/Button.tsx:19]: then all these codes could be deleted, and the things from above be used as props here

File: src/components/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    buttonSecondaryBlackSmallDefault: "bg-dark-light text-white rounded font font-700 py-2.5 disabled:opacity-50 ",
+   
+
+  }
+
+  return (
+    
+    <button
+      className={cn(
+        "flex justify-center text-center text-s px-9 w-2 .w-full",
```
Comment (ReallyBoard123) [inline src/components/Button.tsx:30]: w-2 .w-full ?? these buttons will be very small in width

do you just want it to be w-full?

File: src/components/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+>
+
+export function Button({
+  variant = "buttonPrimaryYellowDefault",
+  className,
+  children = "Button",
+  ...props
+}: ButtonProps) {
+  const styles: Record<typeof variant, string> = {
+    buttonPrimaryYellowDefault: "bg-yellow  text-dark-light rounded font font-700 py-4  disabled:opacity-50",
```
Comment (ReallyBoard123) [inline src/components/Button.tsx:17]: i think we have two different roundedness. On default it is 8px and on the small one it is 6px

**Review by ReallyBoard123 — CHANGES_REQUESTED** [review]
The code is almost there. Just a bit of clean up and abstracting the code to make it neat! But overall good job. You would also need to update the stories once you find the right settings for your button

**Comment by radicalgraphical** [comment]
Hey shiraq , i pushed the updated button :)

File: src/components/Button.tsx (hunk 1 above)
Comment (JanHendryk) [reply src/components/Button.tsx:13]: Done!

File: src/components/Button.tsx (hunk 2 above)
Comment (JanHendryk) [reply src/components/Button.tsx]: Yeah, done!

File: src/components/Button.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import { cn } from "../lib/utils"
+
+type ButtonProps = {
```
[the remark this answers is not in the document]
Comment (JanHendryk) [reply src/components/Button.tsx:6]: Done, we changed it to a more abstract component

**Review by ReallyBoard123 — APPROVED** [review]

**Pull request merged**